lib/explorer/series.ex

```diff
@@ -924,28 +924,30 @@ defmodule Explorer.Series do
   ## Examples
 
       iex> s = [1, 2, 3, 4] |> Explorer.Series.from_list()
-      iex> Explorer.Series.cum_max(s)
+      iex> Explorer.Series.cumulative_max(s)
       #Explorer.Series<
         integer[4]
         [1, 2, 3, 4]
       >
 
       iex> s = [1, 2, nil, 4] |> Explorer.Series.from_list()
-      iex> Explorer.Series.cum_max(s)
+      iex> Explorer.Series.cumulative_max(s)
       #Explorer.Series<
         integer[4]
         [1, 2, nil, 4]
       >
   """
-  @spec cum_max(series :: Series.t(), reverse? :: boolean()) :: Series.t()
-  def cum_max(series, reverse? \\ false)
+  @spec cumulative_max(series :: Series.t(), opts :: Keyword.t()) :: Series.t()
+  def cumulative_max(series, opts \\ [])
 
-  def cum_max(%Series{dtype: dtype} = series, reverse?)
-      when dtype in [:integer, :float, :date, :datetime],
-      do: apply_impl(series, :cum_max, [reverse?])
+  def cumulative_max(%Series{dtype: dtype} = series, opts)
+      when dtype in [:integer, :float, :date, :datetime] do
+    opts = Keyword.validate!(opts, reverse?: false)
```

Comment:
I don't think we use question marks for options (although perhaps we should have):

```suggestion
    opts = Keyword.validate!(opts, reverse: false)
```